File: src/main/java/net/atos/zac/util/ZonedDateTimeAdapter.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.util;

import java.time.ZonedDateTime;
import java.time.format.DateTimeFormatter;

import jakarta.json.bind.adapter.JsonbAdapter;

public class ZonedDateTimeAdapter implements JsonbAdapter<ZonedDateTime, String> {

    @Override
    public String adaptToJson(final ZonedDateTime dateTime) {
        return dateTime != null ? dateTime.format(DateTimeFormatter.ISO_INSTANT) : null;
    }

    @Override
    public ZonedDateTime adaptFromJson(final String dateTime) {
        return ZonedDateTime.parse(dateTime);
    }
}

File: src/main/kotlin/nl/info/zac/util/time/ZonedDateTimeAdapter.kt
/*
 * SPDX-FileCopyrightText: 2021 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.zac.util.time

import jakarta.json.bind.adapter.JsonbAdapter
import java.time.ZonedDateTime
import java.time.format.DateTimeFormatter

class ZonedDateTimeAdapter : JsonbAdapter<ZonedDateTime, String> {
    override fun adaptToJson(dateTime: ZonedDateTime?): String? = dateTime?.format(DateTimeFormatter.ISO_INSTANT)

    override fun adaptFromJson(dateTime: String): ZonedDateTime = ZonedDateTime.parse(dateTime)
}
